pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionRuntimeManager.java

```diff
@@ -440,8 +450,19 @@ public synchronized void restartFunctionInstances(String tenant, String namespac
                         }
                         continue;
                     }
-                    this.functionAdmin.functions().restartFunction(tenant, namespace, functionName,
-                                assignment.getInstance().getInstanceId());
+
+                    ComponentType componentType = assignment.getInstance().getFunctionMetaData().getFunctionDetails().getComponentType();
```

Comment:
the code here and on line 422 - 429 are highly similar.

Could you please abstract a method and then reuse it for these two places.